build-scripts/build-csdk.js

```diff
@@ -75,9 +87,20 @@ if ( fs.existsSync( repoPaths.installPrefix ) ) {
 tinycborPath = path.join( repoPaths.iotivity, "extlibs", "tinycbor", "tinycbor" );
 
 // Attempt to determine the architecture
-architecture = ( process.env.npm_config_user_agent || "" ).match( /\S+/g ) || [];
-architecture =
-	archMap[ architecture[ 2 ] ] ? archMap[ architecture[ 2 ] ][ architecture[ 3 ] ] : null;
+userAgentParts = ( process.env.npm_config_user_agent || "" ).match( /\S+/g ) || [];
+platform = userAgentParts[ 2 ];
+localArch = userAgentParts[ 3 ];
+
+targetArch =
+	archMap[ platform ] ? archMap[ platform ][ localArch ] : null;
```

Comment:
I did that on purpose for code readability, so that everybody understands what is happening there. Are you very strongly in favor of using ```archMap[ platform ][ serAgentParts[ 3 ] ]```? It doesn't have my favor.